SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/RegisterAutoDestroyActorsTest/RegisterAutoDestroyActorsTest.cpp

```diff
@@ -50,44 +50,40 @@ void ARegisterAutoDestroyActorsTestPart1::PrepareTest()
 					++NumCharactersFound;
 				}
 
-				if (NumCharactersFound == NumCharactersExpected)
-				{
-					FinishStep();
-				}
+				RequireEqual_Int(NumCharactersFound, NumCharactersExpected,
+								 TEXT("Clients should observe the correct number of characters in their world."));
+				FinishStep();
 			},
 			5.0f);
 	}
 
 	{ // Step 3 - Destroy by all servers that have authority
 		AddStep(
-			TEXT("SERVER_ALL_RegisterAutoDestroyActors"), FWorkerDefinition::AllServers,
-			[this]() -> bool {
+			TEXT("SERVER_ALL_RegisterAutoDestroyActors"), FWorkerDefinition::AllServers, nullptr, nullptr,
+			[this](float DeltaTime) {
 				int NumCharactersFound = 0;
-				int NumCharactersExpected = 1;
+				int NumCharactersExpected = 1; // 1, because we filter by authority
+				ACharacter* FoundAuthoritativeCharacter = nullptr;
 				UWorld* World = GetWorld();
-				for (TActorIterator<ACharacter> It(World, ACharacter::StaticClass()); It; ++It)
+				for (TActorIterator<ACharacter> It(World); It; ++It)
 				{
 					if (It->HasAuthority())
 					{
 						++NumCharactersFound;
+						FoundAuthoritativeCharacter = *It;
 					}
 				}
-
-				return NumCharactersFound == NumCharactersExpected;
-			},
-			[this]() {
-				UWorld* World = GetWorld();
-				for (TActorIterator<ACharacter> It(World); It; ++It)
+				if (RequireEqual_Int(NumCharactersFound, NumCharactersExpected,
+									 TEXT("Servers should observe the correct number of characters in their world.")))
 				{
-					if (It->HasAuthority())
-					{
-						AssertTrue(IsValid(*It), FString::Printf(TEXT("Registering ACharacter for destruction: %s"), *GetNameSafe(*It)));
-						RegisterAutoDestroyActor(*It);
-					}
+					AssertTrue(IsValid(FoundAuthoritativeCharacter), FString::Printf(TEXT("Registering ACharacter for destruction: %s"),
+																					 *GetNameSafe(FoundAuthoritativeCharacter)));
+					RegisterAutoDestroyActor(FoundAuthoritativeCharacter);
```

Comment:
Why are we only registering the last FoundAuthorativeCharacter for destroying?

Reply:
there should only be one, which we also check later

Reply:
or I guess earlier

Reply:
Got it.